docs/guide/tutorials/task-guide.md

```diff
@@ -0,0 +1,524 @@
+
+If a "normal" (non-mapped) task depends on a mapped task, Prefect automatically applies a reduce operation to gather the mapped results and pass them to the downstream task.
+
+However, if a mapped task relies on another mapped task, Prefect does not reduce the upstream result. Instead, it connects the nth upstream child to the nth downstream child, creating independent parallel pipelines.
+
+Let's have a look at a simple example:
+
+```python
+import prefect
+from prefect import task, Flow
+
+@task
+def add_one(x):
+    """Operates one number at a time"""
+    return x + 1
+
+@task(name="sum")
+def sum_numbers(y):
+    """Operates on an iterable of numbers"""
+    result = sum(y)
+    logger = prefect.context["logger"]
+    logger.info("The sum is {}".format(result))
+    return result
+
+
+## Imperative API
+flow = Flow("Map Reduce")
+flow.set_dependencies(add_one, keyword_tasks={"x": [1, 2, 3, 4]}, mapped=True)
+flow.set_dependencies(sum_numbers, keyword_tasks={"y": add_one})
+
+## Functional API
+with Flow("Map Reduce") as flow:
+    mapped_result = add_one.map(x=[1, 2, 3, 4])
+    summed_result = sum_numbers(mapped_result)
+```
+
+Whenever this Flow is run, we will see the following log:
+```
+[2019-07-20 21:35:00,968] - INFO - prefect.Task | The sum is 14
+```
+
+Note that each "child" task is a _first class Prefect Task_.  This means that they can do anything a "normal" task can do, including succeed, fail, retry, pause, or skip.
+
+#### Not Everything must be mapped over
+
+Suppose we have a Task that accepts many keyword arguments and we want to only map over a _subset_ of those arguments.  In this case, we can use Prefect's `unmapped` container for specifying those inputs which should _not_ be mapped over:
+
+```python
+from prefect import task, Flow
+from prefect.utilities.tasks import unmapped
+
+@task
+def add_one(x, y):
+    """Operates one number at a time"""
+    return x + y
+
+with Flow("unmapped example") as flow:
+    result = add_one.map(x=[1, 2, 3], y=unmapped(5))
+```
+When this Flow runs, only the `x` keyword will be mapped over; the `y` will remain fixed with the constant value of 5.
+
+#### Mapped Tasks don't have to exchange data
+Prefect's API is designed for maximum flexibility - it is actually possible to spawn multiple mapped copies of a Task in response to an upstream output without exchanging data.  For example:
+
+```python
+from prefect import task, Flow
+
+@task
+def return_list():
+    return [1, 2, 3, 4]
+
+@task
+def print_hello():
+    print("=" * 30)
+    print("HELLO")
+    print("=" * 30)
+
+with Flow("print-example") as flow:
+    result = print_hello.map(upstream_tasks=[return_list])
```

Comment:
Worth to note what happens on a map when the upstream returns an empty iterable?

Reply:
Definitely worth a note.